web/modules/custom/dbcdk_community/src/Profile/ProfileRepository.php

```diff
@@ -0,0 +1,357 @@
+   *   The username for the profile to retrieve.
+   *
+   * @return Profile|NULL
+   *   The profile corresponding to the username. NULL if no profile is found.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   */
+  public function getProfileByUsername($username) {
+    $filter = [
+      'where' => ['username' => $username],
+    ];
+    return $this->findProfile($filter);
+  }
+
+  /**
+   * Retrieve a profile matching a filter with community roles and quarantines.
+   *
+   * @param array $filter
+   *   The filter to use for retrieving the profile.
+   *
+   * @return Profile|NULL
+   *   The profile corresponding to the filter. NULL if no profiles match.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   */
+  protected function findProfile(array $filter) {
+    $profile = $this->profileApi->profileFindOne($this->processFilter($filter));
+    if (!empty($profile)) {
+      $profile = $this->enrichProfile($profile);
+    }
+    return $profile;
+  }
+
+  /**
+   * Get all profiles matching a filter.
+   *
+   * @param array $filter
+   *   The filter to use.
+   *
+   * @return \DBCDK\CommunityServices\Model\Profile[]
+   *   The matching profiles.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
+   *   Throws API Exception if any of the calls to the services fails.
+   */
+  public function getProfiles(array $filter = []) {
+    return (array) $this->profileApi->profileFind($this->processFilter($filter));
+  }
+
+  /**
+   * Count the number of profiles matching a filter.
+   *
+   * @param array $filter
+   *   The filter to use.
+   *
+   * @return int|NULL
+   *   The number of profiles matching the filter. NULL if no count is returned.
+   */
+  public function countProfiles(array $filter = []) {
+    $result = $this->profileApi->profileCount($this->processFilter($filter));
+    return (isset($result['count'])) ? $result['count'] : NULL;
+  }
+
+  /**
+   * Enrich a "flat" profile object.
+   *
+   * This adds details about community roles and quarantines.
+   *
+   * @param ModelProfile $profile
+   *   The original profile object.
+   *
+   * @return Profile
+   *   The profile object enriched with details ab
+   */
+  public function enrichProfile(ModelProfile $profile) {
+    $profile = new Profile($profile);
+    $profile->setCommunityRoles((array) $this->profileApi->profilePrototypeGetCommunityRoles($profile->getId()));
+    $profile->setQuarantines((array) $this->profileApi->profilePrototypeGetQuarantines($profile->getId()));
+    return $profile;
+  }
+
+  /**
+   * Get profiles with an active quarantine at a specific time.
+   *
+   * @param DateTime $time
+   *   The time to check quarantine date ranges against.
+   * @param array $profiles_filter
+   *   An array containing filter arguments.
+   *
+   * @throws \DBCDK\CommunityServices\ApiException
```

Comment:
In most (if not all) of our DocBlocks we document `@throws` below `@return`. An example would be the method below this one (countQuarantinedProfiles). CodeSniffer and Scrutinizer will not complain about this - but my OCD kicks in a bit :smiley: